Sources/StreamChatUI/ChatChannel/ChatChannelVC.swift

```diff
@@ -0,0 +1,215 @@
+    /// User search controller passed directly to the composer
+    open var userSuggestionSearchController: ChatUserSearchController!
+
+    /// Controller for observing data changes within the channel
+    open var channelController: ChatChannelController!
+
+    public var client: ChatClient {
+        channelController.client
+    }
+
+    /// Component responsible for setting the correct offset when keyboard frame is changed.
+    open lazy var keyboardHandler: KeyboardHandler = ComposerKeyboardHandler(
+        composerParentVC: self,
+        composerBottomConstraint: messageComposerBottomConstraint
+    )
+
+    open lazy var messageListVC: ChatMessageListVC = components
+        .messageListVC
+        .init()
+
+    /// Controller that handles the composer view
+    open private(set) lazy var messageComposerVC = components
+        .messageComposerVC
+        .init()
+
+    private var messageComposerBottomConstraint: NSLayoutConstraint?
+
+    /// Header View
+    open private(set) lazy var headerView: ChatChannelHeaderView = components
+        .channelHeaderView.init()
+        .withoutAutoresizingMaskConstraints
+
+    /// View for displaying the channel image in the navigation bar.
+    open private(set) lazy var channelAvatarView = components
+        .channelAvatarView.init()
+        .withoutAutoresizingMaskConstraints
+
+    override open func setUp() {
+        super.setUp()
+
+        messageListVC.delegate = self
+        messageListVC.dataSource = self
+        messageListVC.client = client
+
+        messageComposerVC.channelController = channelController
+        messageComposerVC.userSearchController = userSuggestionSearchController
+
+        channelController.setDelegate(self)
+        channelController.synchronize { [weak self] _ in
+            self?.messageComposerVC.updateContent()
+        }
+    }
+
+    override open func setUpLayout() {
+        super.setUpLayout()
+
+        view.backgroundColor = appearance.colorPalette.background
+
+        messageListVC.view.translatesAutoresizingMaskIntoConstraints = false
+        addChildViewController(messageListVC, targetView: view)
+        messageListVC.view.pin(anchors: [.top, .leading, .trailing], to: view.safeAreaLayoutGuide)
+
+        messageComposerVC.view.translatesAutoresizingMaskIntoConstraints = false
+        addChildViewController(messageComposerVC, targetView: view)
+        messageComposerVC.view.pin(anchors: [.leading, .trailing], to: view)
+        messageComposerVC.view.topAnchor.pin(equalTo: messageListVC.view.bottomAnchor).isActive = true
+        messageComposerBottomConstraint = messageComposerVC.view.bottomAnchor.pin(equalTo: view.bottomAnchor)
+        messageComposerBottomConstraint?.isActive = true
+
+        NSLayoutConstraint.activate([
+            channelAvatarView.widthAnchor.pin(equalTo: channelAvatarView.heightAnchor),
+            channelAvatarView.heightAnchor.pin(equalToConstant: 32)
```

Comment:
magic number 32 should be moved into 
```Swift
private enum Dimensions {
    static let channelAvatarHeight: CGFloat = 32
}
```

and used as 
`channelAvatarView.heightAnchor.pin(equalToConstant: Dimensions.channelAvatarHeight)`

https://github.com/raywenderlich/swift-style-guide#constants

Reply:
Done ✅